File: iCovieApi/iCovieApi/Models/Master/RoleModel.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace iCovieApi.Models
{
    public class RoleModel
    {
        public int id { get; set; }
        public string RoleName { get; set; }
        public string RoleShortName { get; set; }
        //public bool Approver { get; set; }
        public List<MainMenuModel> permissions { get; set; }
        //public List<OrgModel> orgpermissions { get; set; }
        public string isCoreRole { get; set; }

        public Boolean isadmin { get; set; }

        public int privilegeid { get; set; }

        public int companyid { get; set; }

        public int locationid { get; set; }
        public int mainmenuid { get; set; }
        public int roleid { get; set; }
        public int mainmenumappingflag { get; set; }
    }
}

File: iCovieAdmin-Ng/src/app/core/auth/_models/MAdministrator/role.model.ts


export class Role {
    id: number;
    RoleName: string;
    RoleShortName: string;
    permissions: any[];
    //orgpermissions: any[];
    isCoreRole: boolean = false;
    privilegeid:number;
    mainmenuid:number;
    roleid:number;
    companyid:number;
    companyids:number[];
    companyname:string;
    locationid:number;
    locationids:number[];
    locationname:string;
    isadmin:boolean;   
    issuperadmin:boolean;   
    
    clear(): void {
        this.id = undefined;
        this.RoleName = '';
        this.RoleShortName = '';
        this.permissions = [];       
        this.isCoreRole = false;
        this.privilegeid=undefined;
        this.mainmenuid=undefined;
        this.roleid=undefined;
        this.isadmin=false;
        this.issuperadmin = false;
        this.companyid =undefined;
        this.companyname= '';
        this.locationid= undefined;
        this.locationname= '';
	}
}
